configs/raspberrypi_config

```diff
@@ -78,25 +83,40 @@ function platform_modify() {
         run_and_log_cmd "sudo cp work/boot/${RASPBERRYPI_KERNEL_IMAGE} work/rootfs/boot/${MENDER_KERNEL_IMAGETYPE}"
     fi
 
+    # Make sure the on-board bootloader boots our bootloader (which we do via the kernel option).
+    sed -i -e '/^kernel=/d' work/boot/config.txt
+    echo "kernel=${RASPBERRYPI_BOOTLOADER_IMAGE}" >> work/boot/config.txt
+
     # Replace kernel with U-boot and add boot script
     run_and_log_cmd "sudo mkdir -p work/rootfs/uboot"
     run_and_log_cmd "sudo cp work/rpi/binaries/u-boot.bin work/boot/${RASPBERRYPI_BOOTLOADER_IMAGE}"
     run_and_log_cmd "sudo cp work/rpi/binaries/boot.scr work/boot"
     run_and_log_cmd "sudo cp work/rpi/binaries/fw_env.config work/rootfs/etc/"
     run_and_log_cmd "sudo cp work/rpi/binaries/uboot-git-log.txt work/boot"
 
+    # Disable 64-bit mode and force 32-bit mode. Even though we check for 32-bit binaries at the
+    # start of this function, some images use a 64-bit kernel even with 32-bit userspace. Due to our
+    # custom U-Boot bootloader we need to stick to 32-bit for now.
+    if ! grep -q '^arm_64bit=0' work/boot/config.txt; then
+        log_info "Forcing 32-bit kernel"
+        sed -i -e '/^arm_64bit=/d' work/boot/config.txt
+        echo "arm_64bit=0" >> work/boot/config.txt
+    fi
+
     # Raspberry Pi applications expect to find this on the device and in some
     # cases parse the options to determine the functionality. Create symlinks
-    # to all files
+    # to all files that don't already exist.
     for f in $(ls -1 work/boot); do
```

Comment:
f should be a local variable in this scope. and you should traverse the files via:
```bash
while read -r f; do 
 # now remember that "$f" contains the full path
 [[ -e "$f" ]] && run_and_log_cmd "sudo ln -fs /uboot/`basename ${f}` work/rootfs/boot/${f}"
 # I do not uderstand ln command, see below comment.
done < <(find work/boot -type f)
```
`-e` is on purpose there? I am so used to `-f` :)

Reply:
`f` is local when it is for-loop-initialized, isn't it?

`-e` was originally on purpose, but maybe `-f` is better. Then at least we will get an error message if it's a directory.

I'm not sure I get your variant though, what does it fix?

Reply:
I don't think I will do any more followup here, the loop is anyway pre-existing.